stock_account_internal_move/models/stock_move.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2018 by Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _action_done(self):
+        res = super()._action_done()
+        for move in self:
+            if not (move.location_id.usage == 'internal'
+                    and move.location_dest_id.usage == 'internal'):
+                continue
+
+            location_src = move.location_id
+            location_dest = move.location_dest_id
+            if location_src.force_accounting_entries \
+                    != location_dest.force_accounting_entries:
+                raise ValidationError(_(
+                    'Could not generate a journal entry.'
+                    ' You need to either force or disable accounting on both'
+                    ' locations in order to do that.'))
+            if not location_src.force_accounting_entries:
```

Comment:
elif?

Reply:
no use, since previous `if` branch was trying to raise `ValidationError`